### ISSUE TITLE
PM2 Unable to start API with Cluster mode with Interprester

### ISSUE THREAD

**Issue by Mayankpateldev** [Unitech/pm2#4886]
## What's going wrong?

sudo node version
6.10.0
node version
13.14.0

I am using ecosystem.config.js for run my application
     ```
   name: api,
        script: 'bin/www_ssl',
        cwd: projectPath,
        exec_mode: "cluster",
        instances: 4,
       interpreter: "/home/ubuntu/.nvm/versions/node/v13.14.0/bin/node",
        log_date_format: 'YYYY-MM-DD HH:mm:ss.SSS',
        enable: true
```

using below command to run
`sudo pm2 start ecosystem.config.js
`
In desc command I can able to see interpreter path is correct but picking wrong node version rather than choose 13.14.0 version pm2 using older version 6.10.0 version

Can you please help me to solve this as soon as.

## How could we reproduce this issue?
You can easily reproduce this using above config
## Supporting information

```
$ pm2 report
```

**Comment by Shogobg** [comment]
This is a workaround for your problem, i don't know if it can be done with the configuration alone.
Write `which pm2` - this will print the path to your pm2 binary
`cd <path_to_pm2_binary>`
`cp pm2 pm2-node13`
Open pm2-node13 with your favorite text editor (i use Nano)
On the first line you will see this `#!/usr/bin/env node`
Replace it with `#!/home/ubuntu/.nvm/versions/node/v13.14.0/bin node`
After all this is finished, you should be able to use `sudo pm2-node13 start ecosystem.config.js` to start your application.

**Comment by inanibharat** [comment]
@Shogobg.

Thanks for steps but i faced below issue when I implement your steps:

sudo: unable to execute /usr/bin/pm2-node 13: Permission denied

Kindly guide me to resolve that issue.

Thanks again.

**Comment by Shogobg** [comment]
@inanibharat 
I see you have space between node and 13 - make sure you're trying to call the correct script.
Also set the executable flag for the script and the runtime
`chmod +x /usr/bin/pm2-node13`
`chmod +x /<path_to_your_node_js_version_bin_folder>/node`

You can see the privileges with `ls -lah /usr/bin/pm2-node13`
You must have +x.

This may not be needed, but as a last precaution you can change the owner
`chown your_username:your_group /usr/bin/pm2-node13`
`chown your_username:your_group /<path_to_your_node_js_version_bin_folder>/node`

**Comment by inanibharat** [comment]
Hi,

Still i am facing issue.

ubuntu@ip-172-31-54-101:~$ sudo pm2-node13 
sudo: unable to execute /usr/bin/pm2-node13: Permission denied
ubuntu@ip-172-31-54-101:~$ ll /usr/bin/pm2-node13
-rwxrwxrwx 1 root root 88 Nov  4 10:14 /usr/bin/pm2-node13*
ubuntu@ip-172-31-54-101:~$ chmod +x /usr/bin/pm2-node13
chmod: changing permissions of '/usr/bin/pm2-node13': Operation not permitted
ubuntu@ip-172-31-54-101:~$ sudo chmod +x /usr/bin/pm2-node13
ubuntu@ip-172-31-54-101:~$ sudo pm2-node13 
sudo: unable to execute /usr/bin/pm2-node13: Permission denied
ubuntu@ip-172-31-54-101:~$ pm2-node13 
-bash: /usr/bin/pm2-node13: /home/ubuntu/.nvm/versions/node/v13.11.0/bin: bad interpreter: Permission denied

**Comment by inanibharat** [comment]
Hi @Shogobg ,

Still i am facing same issue. still not able to run cluster mode for pm2 in node 13.

Permission also assigned.

It would be appreciated if you will be guide us to resolve issue. 

Thanks,
Bharat Prakash Inani

**Comment by Shogobg** [comment]
Hello @inanibharat 

I don't know your exact setup, so i'm just guessing what it could be.
1. Less likely, your file system was not mounted correctly - https://stackoverflow.com/questions/15214890/unable-to-execute-script-file-with-x-permission-even-with-sudo
2. More likely, your account doesn't have permissions to sudo execute node.js - https://www.linuxquestions.org/questions/linux-newbie-8/how-to-fix-sudo-unable-to-execute-bin-ls-permission-denied-4175426455/
3. Another reason unique to your case

What you can try:
1. Change the owner of the files(Nodejs and PM2) from root to Ubuntu(your user), this way you don't have to use Sudo
2. Try running them again with ubuntu.

**Comment by Tomas2D** [comment]
@Shogobg I think that your solution does not work. When I edited my application to log node version on startup (`console.info(process.version)`) I am still getting the default version of the node which pm2 uses. Did you notice that?

**Comment by Shogobg** [comment]
@Tomas2D , did you resolve your issue?
If not, let me know what command you use to run your application.
Everyone's setup might be different than the other, so what works for me or Bharat might not work for you.

**Comment by Tomas2D** [comment]
@Shogobg nope. What I did is that I upgraded my apps running in cluster mode to Node 16.

**Comment by rafal-ksiazek-rmtcfm-com** [comment]
I had the same issue, but the /usr/bin/pm2 is symlink to ../lib/node_modules/pm2/bin/pm2

And if you just copy it inside the /usr/bin folder then this line will not work:
`require('../lib/binaries/CLI.js');`

So to run this we have to:
1) go to pm2 location
`cd /usr/lib/node_modules/pm2/bin/pm2`
2) copy pm2
`cp pm2 pm2-node16`
3) go to /usr/bin
`cd /usr/bin`
4) create symlink
`ln -nfs ../lib/node_modules/pm2/bin/pm2-node16 .`
5) change it
`nano pm2-node16`
6) and replace 
`#!/usr/bin/env node` => `#!/home/YOUR_USER/.nvm/versions/node/v16.20.2/bin/node`

Now we can start our project:
`PM2_HOME='.pm2-node16' pm2-node16 start pm2.json` 

But in cluster mode it's still using default server nodejs 14.21.3. 

Because the daemon is using default server nodejs. I don't know how to force daemon to use different one.
